compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/input/key/KeyEvent.uikit.kt

```diff
@@ -24,28 +24,52 @@ import platform.UIKit.UIKeyModifierShift
 import platform.UIKit.UIPress
 import platform.UIKit.UIPressPhase.UIPressPhaseBegan
 import platform.UIKit.UIPressPhase.UIPressPhaseEnded
+import platform.UIKit.UIPressTypeDownArrow
+import platform.UIKit.UIPressTypeLeftArrow
+import platform.UIKit.UIPressTypePageDown
+import platform.UIKit.UIPressTypePageUp
+import platform.UIKit.UIPressTypeRightArrow
+import platform.UIKit.UIPressTypeUpArrow
 
 internal fun UIPress.toComposeEvent(): KeyEvent {
-    // TODO: https://developer.apple.com/documentation/uikit/uipress/3526315-key
-    //  can be potentially nil on TVOS, this will cause a crash
-    val uiKey = requireNotNull(key) {
-        "UIPress with null key is not supported"
+    val pressKey = key
+
+    val keyEventType = when (phase) {
+        UIPressPhaseBegan -> KeyEventType.KeyDown
+        UIPressPhaseEnded -> KeyEventType.KeyUp
+        else -> KeyEventType.Unknown
+    }
+
+    // UIPress has special types for arrow keys and page up/down and has null `key`
+    // In other cases the `type` returns an int value that doesn't match any UIPressType.
+    val specialTypeKey = when (type) {
+        UIPressTypeUpArrow -> Key.DirectionUp
+        UIPressTypeDownArrow -> Key.DirectionDown
+        UIPressTypeLeftArrow -> Key.DirectionLeft
+        UIPressTypeRightArrow -> Key.DirectionRight
+        UIPressTypePageDown -> Key.PageDown
+        UIPressTypePageUp -> Key.PageUp
+        else -> null
     }
+
+    val key = specialTypeKey ?: pressKey?.keyCode?.let { Key(it) } ?: Key.Unknown
+    val codePoint = pressKey?.characters?.firstOrNull()?.code ?: 0
```

Comment:
Please reuse our internal helper `String.codePointAt(index: Int): CodePoint` instead of `firstOrNull`

Reply:
I feel hesitant to introduce this change:
1. I'm not owning this piece of code.
2. This is out of the scope of the fix.
3. So far we have no issues related to it.

Reply:
Let's add explicit TODO at least

Reply:
Done